base/sparse/sparsematrix.jl

```diff
@@ -93,11 +91,15 @@ function Base.showarray(io::IO, S::SparseMatrixCSC;
     pad = ndigits(max(S.m,S.n))
     k = 0
     sep = "\n\t"
```

Comment:
Yes, we currently don't seem to have a way to print sparse matrices on one line. Developing such a format is out of scope here.